pkgs/development/libraries/bashup-events/generic.nix

```diff
@@ -0,0 +1,90 @@
+{
+  # general
+  lib
+, callPackage
+, runCommand
+, resholvePackage
+, fetchFromGitHub
+, bash
+, shellcheck
+, doCheck ? true
+, doInstallCheck ? true
+  # variant-specific
+, variant
+, version
+, branch
+, rev
+, hash
```

Comment:
It would be simpler to pass the `pname` and `src`. It's a bit of duplication but it would also make things more generic.

Reply:
Hmm. I don't mind passing the src if you think it is an improvement (done in https://github.com/NixOS/nixpkgs/pull/107182/commits/36d15d622ae29127157a23b3faf01f6757e4960b).

But I think pre-flattening the "variant" down into the pname undercuts my objective of keeping things at least a little weird to emphasize that this package has an odd naming convention (it contains a thing that looks like version number, but should not be treated as the version number of this package).

For context, here are the author's comments:
> The main and 4.4 branches are both valid and useful things to have, as each has specialty semantics that are useful in different contexts. I don't really plan to deprecate the 3.x version, even though it may move off the main branch at some point in the future. And if at some point they do have official versions, they'd be something like bashup/events-3 1.0 and bashup/events-4 1.0. :)